## Librerias

In [6]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None)
#pd.set_option('display.max_rows', None)

## Carga de datos

In [2]:
s10 = pd.read_csv("C:\\Users\\camil\\OneDrive\\Documentos\\GitHub\\GITHUb\\EDA_DS\\Src\\Data\\raw\\08_19_2020.csv", sep=";")

## Primer análisis

In [3]:
s10

gameId league split  playoffs              date  game  patch  \
0     1241318  KeSPA   NaN         0   03/01/2020 7:33     1   9.24   
1     1241322  KeSPA   NaN         0   03/01/2020 9:00     2   9.24   
2     1241324  KeSPA   NaN         0  03/01/2020 10:05     3   9.24   
3     1241328  KeSPA   NaN         0  03/01/2020 10:57     4   9.24   
4     1241386  KeSPA   NaN         0   04/01/2020 5:33     1   9.24   
...       ...    ...   ...       ...               ...   ...    ...   
5144  1310259     EM   NaN         0  18/08/2020 20:25     1  10.16   
5145  1310260     EM   NaN         0  18/08/2020 20:27     1  10.16   
5146  1310262     EM   NaN         0  18/08/2020 20:53     1  10.16   
5147  1320317     EM   NaN         0  18/08/2020 20:56     1  10.16   
5148  1310264     EM   NaN         0  18/08/2020 21:27     1  10.16   

      gamelength          t1_playerid   t1_ban1       t1_ban2   t1_ban3  \
0           2220       SANDBOX Gaming   LeBlanc        Irelia   Rek'Sai   
1           2227                   T1    Syndra       LeBlanc    Rumble   
2           1711                   T1   LeBlanc        Rumble   Lee Sin   
3           1999                   T1    Rumble         Varus   LeBlanc   
4           1812                  DRX    Qiyana       Orianna    Irelia   
...          ...                  ...       ...           ...       ...   
5144        2075               Racoon   Caitlyn         Karma       Lux   
5145        2095      SINNERS Esports      Shen         Senna      Azir   
5146        2879     Team GamerLegion   Nidalee  Twisted Fate      Gnar   
5147        1874            G2 Arctic  Cho'Gath          Sett  Renekton   
5148        1808  Intrepid Fox Gaming     Akali       Camille     Senna   

           t1_ban4    t1_ban5  t1_result  t1_firstblood  t1_firstdragon  \
0            Yasuo   Renekton          1            0.0             0.0   
1            Braum      Leona          0            0.0             0.0   
2           Gragas       Olaf          1            0.0             1.0   
3           Aatrox     Syndra          0            0.0             1.0   
4       Tahm Kench  Gangplank          0            0.0             0.0   
...            ...        ...        ...            ...             ...   
5144          Shen   Malphite          1            0.0             1.0   
5145  Twisted Fate    Taliyah          1            0.0             0.0   
5146          Olaf       Azir          1            1.0             0.0   
5147   Mordekaiser       Shen          1            0.0             0.0   
5148           Zoe    LeBlanc          1            0.0             1.0   

      t1_dragons  t1_elementaldrakes  t1_infernals  t1_mountains  t1_clouds  \
0              2                 2.0           0.0           2.0        0.0   
1              1                 1.0           0.0           0.0        0.0   
2              4                 4.0           0.0           2.0        1.0   
3              2                 2.0           0.0           1.0        0.0   
4              1                 1.0           0.0           0.0        0.0   
...          ...                 ...           ...           ...        ...   
5144           4                 4.0           2.0           1.0        0.0   
5145           2                 2.0           0.0           0.0        0.0   
5146           4                 4.0           0.0           3.0        0.0   
5147           2                 2.0           1.0           1.0        0.0   
5148           4                 4.0           0.0           2.0        1.0   

      t1_oceans  t1_elders  t1_firstherald  t1_heralds  t1_firstbaron  \
0           0.0        0.0             1.0         1.0            1.0   
1           1.0        0.0             0.0         0.0            0.0   
2           1.0        0.0             1.0         2.0            1.0   
3           1.0        0.0             1.0         1.0            0.0   
4           1.0        0.0            

In [36]:
s10.info(verbose=True, show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5149 entries, 0 to 5148
Data columns (total 408 columns):
 #    Column                        Non-Null Count  Dtype  
---   ------                        --------------  -----  
 0    gameId                        5148 non-null   object 
 1    league                        5149 non-null   object 
 2    split                         3566 non-null   object 
 3    playoffs                      5149 non-null   int64  
 4    date                          5149 non-null   object 
 5    game                          5149 non-null   int64  
 6    patch                         5149 non-null   float64
 7    gamelength                    5149 non-null   int64  
 8    t1_playerid                   5149 non-null   object 
 9    t1_ban1                       4839 non-null   object 
 10   t1_ban2                       4843 non-null   object 
 11   t1_ban3                       5116 non-null   object 
 12   t1_ban4                       4833 non-null   

In [37]:
s10.describe(exclude=object).round(2)

playoffs     game    patch  gamelength  t1_result  t1_firstblood  \
count   5149.00  5149.00  5149.00     5149.00    5149.00        5148.00   
mean       0.09     1.51    10.09     1954.63       0.53           0.51   
std        0.28     0.79     0.06      346.56       0.50           0.50   
min        0.00     1.00     9.24     1005.00       0.00           0.00   
25%        0.00     1.00    10.06     1711.00       0.00           0.00   
50%        0.00     1.00    10.11     1913.00       1.00           1.00   
75%        0.00     2.00    10.13     2158.00       1.00           1.00   
max        1.00     5.00    10.16     3751.00       1.00           1.00   

       t1_firstdragon  t1_dragons  t1_elementaldrakes  t1_infernals  \
count         5095.00     5149.00             5067.00       5067.00   
mean             0.47        2.45                2.32          0.59   
std              0.50        1.48                1.37          0.80   
min              0.00        0.00                0.00          0.00   
25%              0.00        1.00                1.00          0.00   
50%              0.00        3.00                2.00          0.00   
75%              1.00        4.00                4.00          1.00   
max              1.00       10.00                4.00          4.00   

       t1_mountains  t1_clouds  t1_oceans  t1_elders  t1_firstherald  \
count       5067.00    5067.00    5067.00    5098.00         5098.00   
mean           0.57       0.57       0.60       0.08            0.60   
std            0.81       0.79       0.82       0.28            0.49   
min            0.00       0.00       0.00       0.00            0.00   
25%            0.00       0.00       0.00       0.00            0.00   
50%            0.00       0.00       0.00       0.00            1.00   
75%            1.00       1.00       1.00       0.00            1.00   
max            4.00       4.00       4.00       2.00            1.00   

       t1_heralds  t1_firstbaron  t1_barons  t1_firsttower  t1_towers  \
count     5015.00        4964.00    5149.00        5119.00    5149.00   
mean         1.18           0.49       0.64           0.57       6.23   
std          0.77           0.50       0.69           0.50       3.58   
min          0.00           0.00       0.00           0.00       0.00   
25%          1.00           0.00       0.00           0.00       3.00   
50%          1.00           0.00       1.00           1.00       7.00   
75%          2.00           1.00       1.00           1.00       9.00   
max          2.00           1.00       4.00           1.00      11.00   

       t1_firstmidtower  t1_firsttothreetowers  t1_inhibitors  t2_result  \
count           5091.00                5098.00        5098.00    5149.00   
mean               0.57                   0.56           1.00       0.47   
std                0.50                   0.50           1.12       0.50   
min                0.00                   0.00           0.00       0.00   
25%                0.00                   0.00           0.00       0.00   
50%                1.00                   1.00           1.00       0.00   
75%                1.00                   1.00           2.00       1.00   
max                1.00                   1.00           7.00       1.00   

       t2_firstblood  t2_firstdragon  t2_dragons  t2_elementaldrakes  \
count        5148.00         5095.00     5149.00             5067.00   
mean            0.49            0.53        2.45                2.31   
std             0.50            0.50        1.55                1.39   
min             0.00            0.00        0.00                0.00   
25%             0.00            0.00        1.00                1.00   
50%             0.00            1.00        2.00                2.00   
75%             1.00            1.00        4.00                4.00   
max             1.00            1.00       10.00                4.00   

       t2_infernals  t2_mountains  t2_clouds  t2_o

In [38]:
s10.describe(exclude=object).shape

(8, 352)

In [39]:
s10.describe(include=object)

gameId league   split              date t1_playerid  t1_ban1 t1_ban2  \
count      5148   5149    3566              5149        5149     4839    4843   
unique     5148     23       7              5027         244      107     109   
top     1241318    LDL  Spring  06/04/2020 19:28         DRX  Kalista    Sett   
freq          1    840    1522                 4          59      314     266   

       t1_ban3  t1_ban4  t1_ban5      t2_playerid t2_ban1 t2_ban2 t2_ban3  \
count     5116     4833     4821             5149    5115    5118    5111   
unique     111      128      124              244     101     105     106   
top       Sett  LeBlanc  LeBlanc  FunPlus Phoenix   Varus    Sett    Sett   
freq       348      286      230               59     810     427     441   

        t2_ban4  t2_ban5 t1p1_position t1p1_player t1p1_team t1p1_champion  \
count      5102     5090          5149        5149      5149          5149   
unique      135      125             1         341       244            51   
top     LeBlanc  LeBlanc           sup       Keria       DRX      Nautilus   
freq        268      243          5149          59        59           946   

       t1p2_position t1p2_player t1p2_team t1p2_champion t1p3_position  \
count           5149        5149      5149          5149          5149   
unique             1         331       244            54             1   
top              bot        Deft       DRX      Aphelios           mid   
freq            5149          59        59          1184          5149   

       t1p3_player t1p3_team t1p3_champion t1p4_position t1p4_player  \
count         5149      5149          5149          5149        5149   
unique         349       244            83             1         335   
top          Chovy       DRX        Syndra           jng      Pyosik   
freq            57        59           555          5149          59   

       t1p4_team t1p4_champion t1p5_position t1p5_player t1p5_team  \
count       5149          5149          5149        5149      5149   
unique       244            47             1         339       244   
top          DRX       Lee Sin           top       Doran       DRX   
freq          59           720          5149          57        59   

       t1p5_champion t2p1_position t2p1_player        t2p1_team t2p1_champion  \
count           5149          5149        5149             5149          5149   
unique            72             1         347              244            55   
top             Ornn           sup      Shadow  FunPlus Phoenix      Nautilus   
freq             735          5149          60               59           930   

       t2p2_position t2p2_player        t2p2_team t2p2_champion t2p3_position  \
count           5149        5149             5149          5149          5149   
unique             1         335              244            54             1   
top              bot         Lwx  FunPlus Phoenix      Aphelios           mid   
freq            5149          58               59           910          5149   

       t2p3_player        t2p3_team t2p3_champion t2p4_position t2p4_player  \
count         5149             5149          5149          5149        5149   
unique         344              244            87             1         331   
top          Doinb  FunPlus Phoenix           Zoe           jng        Tian   
freq            59               59           529          5149          59   

              t2p4_team t2p4_champion t2p5_position t2p5_player  \
count              5149          5149          5149        5149   
unique              244            47             1         334   
top     FunPlus Phoenix       Lee Sin           top  StarScreen   
freq                 59           778          5149          50   

              t2p5_team t2p5_champion  
count              5149          5149  
unique              244            70  
top     FunPlus Phoenix          Ornn  
freq                 59           727

In [40]:
s10.describe(include=object).shape

(4, 56)

In [41]:
s10.duplicated().sum()

0

In [42]:
for i in s10.isna().sum()/len(s10)*100:
    print(i)

0.019421246844047387
0.0
30.743833754127014
0.0
0.0
0.0
0.0
0.0
0.0
6.02058652165469
5.9429015342785005
0.6409011458535638
6.137114002718975
6.370168964847543
0.0
0.019421246844047387
1.048747329578559
0.0
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
0.9904835890464169
0.9904835890464169
2.60244707710235
3.5929306661487663
0.0
0.5826374053214216
0.0
1.1264323169547485
0.9904835890464169
0.9904835890464169
0.0
0.6603223926976112
0.602058652165469
0.7380073800738007
0.9127986016702273
1.1458535637987959
0.0
0.019421246844047387
1.048747329578559
0.0
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
1.5925422412118857
0.9904835890464169
0.9904835890464169
2.60244707710235
3.5929306661487663
0.0
0.5826374053214216
0.0
1.1264323169547485
0.9904835890464169
0.9904835890464169
0.0
0.0
0.0
0.0
0.0
0.0
0.0
2.0392309186249755
0.602058652165469
1.009904835890464
1.009904835890464
0.019421246844047387
0.019421246844047387


Tras este primer análisis inicial, el dataset consta de 5149 fias y 408 columnas y ninguna fila con datos incoherentes tanto en el encabezado como en la cola. Se observa que la mayoría de las columnas poseen valores nulos (``Nan``), pero es cierto que ninguna de estas supera el 2% de valores nulos en su interior. Los tipo de dato de las columnas se reparten en 352 numéricas (241 de tipo ``float`` y 61 de tipo ``int``). el tipo de los datos de las 56 columnas restantes es tipo ``objeto``. Además, la información aportada por la función ``describe`` de ``pandas`` nos permite intuir que los datos no siguen una distribución normal aparente. Esta misma función nos puede dar a conocer variables binomiales como ``t1_result`` pues su desviación típica, máximo y mínimo se asemejana a como son los de balores booleanos. Por último, destacamos que no existen columnas duplicadas. 

## Cardinalidad

In [7]:
unique = s10.nunique()
unique

gameId                          5148
league                            23
split                              7
playoffs                           2
date                            5027
game                               5
patch                             16
gamelength                      1403
t1_playerid                      244
t1_ban1                          107
t1_ban2                          109
t1_ban3                          111
t1_ban4                          128
t1_ban5                          124
t1_result                          2
t1_firstblood                      2
t1_firstdragon                     2
t1_dragons                         9
t1_elementaldrakes                 5
t1_infernals                       5
t1_mountains                       5
t1_clouds                          5
t1_oceans                          5
t1_elders                          3
t1_firstherald                     2
t1_heralds                         3
t1_firstbaron                      2
t

In [8]:
for i in s10:
    print(f"La cardinalidad de la variable {i} es {(s10[i].nunique()/s10.shape[0])*100:.2f}%")

La cardinalidad de la variable gameId es 99.98%
La cardinalidad de la variable league es 0.45%
La cardinalidad de la variable split es 0.14%
La cardinalidad de la variable playoffs es 0.04%
La cardinalidad de la variable date es 97.63%
La cardinalidad de la variable game es 0.10%
La cardinalidad de la variable patch es 0.31%
La cardinalidad de la variable gamelength es 27.25%
La cardinalidad de la variable t1_playerid es 4.74%
La cardinalidad de la variable t1_ban1 es 2.08%
La cardinalidad de la variable t1_ban2 es 2.12%
La cardinalidad de la variable t1_ban3 es 2.16%
La cardinalidad de la variable t1_ban4 es 2.49%
La cardinalidad de la variable t1_ban5 es 2.41%
La cardinalidad de la variable t1_result es 0.04%
La cardinalidad de la variable t1_firstblood es 0.04%
La cardinalidad de la variable t1_firstdragon es 0.04%
La cardinalidad de la variable t1_dragons es 0.17%
La cardinalidad de la variable t1_elementaldrakes es 0.10%
La cardinalidad de la variable t1_infernals es 0.10%
La card

In [9]:
cols_card = []
card = []

for i in s10:
    cols_card.append(i)
    card.append((s10[i].nunique()/s10.shape[0])*100)

p100_card = pd.DataFrame(index=cols_card, data=card)
p100_card

0
gameId                        99.980579
league                         0.446689
split                          0.135949
playoffs                       0.038842
date                          97.630608
game                           0.097106
patch                          0.310740
gamelength                    27.248009
t1_playerid                    4.738784
t1_ban1                        2.078073
t1_ban2                        2.116916
t1_ban3                        2.155758
t1_ban4                        2.485920
t1_ban5                        2.408235
t1_result                      0.038842
t1_firstblood                  0.038842
t1_firstdragon                 0.038842
t1_dragons                     0.174791
t1_elementaldrakes             0.097106
t1_infernals                   0.097106
t1_mountains                   0.097106
t1_clouds                      0.097106
t1_oceans                      0.097106
t1_elders                      0.058264
t1_firstherald                 0.038842
t1_heralds                     0.058264
t1_firstbaron                  0.038842
t1_barons                      0.097106
t1_firsttower                  0.038842
t1_towers                      0.233055
t1_firstmidtower               0.038842
t1_firsttothreetowers          0.038842
t1_inhibitors                  0.155370
t2_playerid                    4.738784
t2_ban1                        1.961546
t2_ban2                        2.039231
t2_ban3                        2.058652
t2_ban4                        2.621868
t2_ban5                        2.427656
t2_result                      0.038842
t2_firstblood                  0.038842
t2_firstdragon                 0.038842
t2_dragons                     0.194212
t2_elementaldrakes             0.097106
t2_infernals                   0.097106
t2_mountains                   0.097106
t2_clouds                      0.097106
t2_oceans                      0.097106
t2_elders                      0.077685
t2_firstherald                 0.038842
t2_heralds                     0.058264
t2_firstbaron                  0.038842
t2_barons                      0.097106
t2_firsttower                  0.038842
t2_towers                      0.233055
t2_firstmidtower               0.038842
t2_firsttothreetowers          0.038842
t2_inhibitors                  0.155370
t1p1_position                  0.019421
t1p1_player                    6.622645
t1p1_team                      4.738784
t1p1_champion                  0.990484
t1p1_kills                     0.194212
t1p1_deaths                    0.233055
t1p1_assists                   0.602059
t1p1_firstblood                0.038842
t1p1_firstbloodkill            0.038842
t1p1_firstbloodassist          0.038842
t1p1_firstbloodvictim          0.038842
t1p1_damagetochampions        73.159837
t1p1_dpm                      99.747524
t1p1_wardsplaced               2.447077
t1p1_wpm                      77.937464
t1p1_wardskilled               0.912799
t1p1_wcpm                     63.895902
t1p1_controlwardsbought        0.757429
t1p1_visionscore               3.515246
t1p1_vspm                     83.550204
t1p1_totalgold                63.779375
t1p1_earnedgold               56.243931
t1p1_earned gpm               99.378520
t1p1_goldspent                12.701495
t1p1_minionkills               3.923092
t1p1_monsterkills              0.466110
t1p1_monsterkillsownjungle     0.310740
t1p1_monsterkillsenemyjungle   0.155370
t1p1_cspm                     76.480870
t1p1_goldat10                 21.266265
t1p1pat10                     25.791416
t1p1_csat10                    1.611963
t1p1_goldat15                 30.627306
t1p1pat15                     35.133036
t1p1_csat15                    2.175180
t1p2_position                  0.019421
t1p2_player                    6.428433
t1p2_team                      4.738784
t1p2_champion                  1.048747
t1p2_kills                     0.388425
t1p2_deaths                    0.233055
t1p2_assists                   0.46611

In [10]:
card_0 = pd.concat([unique,p100_card], axis=1, )
card_0

0          0
gameId                        5148  99.980579
league                          23   0.446689
split                            7   0.135949
playoffs                         2   0.038842
date                          5027  97.630608
game                             5   0.097106
patch                           16   0.310740
gamelength                    1403  27.248009
t1_playerid                    244   4.738784
t1_ban1                        107   2.078073
t1_ban2                        109   2.116916
t1_ban3                        111   2.155758
t1_ban4                        128   2.485920
t1_ban5                        124   2.408235
t1_result                        2   0.038842
t1_firstblood                    2   0.038842
t1_firstdragon                   2   0.038842
t1_dragons                       9   0.174791
t1_elementaldrakes               5   0.097106
t1_infernals                     5   0.097106
t1_mountains                     5   0.097106
t1_clouds                        5   0.097106
t1_oceans                        5   0.097106
t1_elders                        3   0.058264
t1_firstherald                   2   0.038842
t1_heralds                       3   0.058264
t1_firstbaron                    2   0.038842
t1_barons                        5   0.097106
t1_firsttower                    2   0.038842
t1_towers                       12   0.233055
t1_firstmidtower                 2   0.038842
t1_firsttothreetowers            2   0.038842
t1_inhibitors                    8   0.155370
t2_playerid                    244   4.738784
t2_ban1                        101   1.961546
t2_ban2                        105   2.039231
t2_ban3                        106   2.058652
t2_ban4                        135   2.621868
t2_ban5                        125   2.427656
t2_result                        2   0.038842
t2_firstblood                    2   0.038842
t2_firstdragon                   2   0.038842
t2_dragons                      10   0.194212
t2_elementaldrakes               5   0.097106
t2_infernals                     5   0.097106
t2_mountains                     5   0.097106
t2_clouds                        5   0.097106
t2_oceans                        5   0.097106
t2_elders                        4   0.077685
t2_firstherald                   2   0.038842
t2_heralds                       3   0.058264
t2_firstbaron                    2   0.038842
t2_barons                        5   0.097106
t2_firsttower                    2   0.038842
t2_towers                       12   0.233055
t2_firstmidtower                 2   0.038842
t2_firsttothreetowers            2   0.038842
t2_inhibitors                    8   0.155370
t1p1_position                    1   0.019421
t1p1_player                    341   6.622645
t1p1_team                      244   4.738784
t1p1_champion                   51   0.990484
t1p1_kills                      10   0.194212
t1p1_deaths                     12   0.233055
t1p1_assists                    31   0.602059
t1p1_firstblood                  2   0.038842
t1p1_firstbloodkill              2   0.038842
t1p1_firstbloodassist            2   0.038842
t1p1_firstbloodvictim            2   0.038842
t1p1_damagetochampions        3767  73.159837
t1p1_dpm                      5136  99.747524
t1p1_wardsplaced               126   2.447077
t1p1_wpm                      4013  77.937464
t1p1_wardskilled                47   0.912799
t1p1_wcpm                     3290  63.895902
t1p1_controlwardsbought         39   0.757429
t1p1_visionscore               181   3.515246
t1p1_vspm                     4302  83.550204
t1p1_totalgold                3284  63.779375
t1p1_earnedgold               2896  56.243931
t1p1_earned gpm               5117  99.378520
t1p1_goldspent                 654  12.701495
t1p1_minionkills               202   3.923092
t1p1_monsterkills               24   0.466110
t1p1_monsterkillsownjungle      16   0.310740
t1p1_monsterkillsenemyjungle     8   0.155370
t1p1_cspm                     3

In [11]:
tipo_datos = s10.dtypes
tipo_datos

gameId                           object
league                           object
split                            object
playoffs                          int64
date                             object
game                              int64
patch                           float64
gamelength                        int64
t1_playerid                      object
t1_ban1                          object
t1_ban2                          object
t1_ban3                          object
t1_ban4                          object
t1_ban5                          object
t1_result                         int64
t1_firstblood                   float64
t1_firstdragon                  float64
t1_dragons                        int64
t1_elementaldrakes              float64
t1_infernals                    float64
t1_mountains                    float64
t1_clouds                       float64
t1_oceans                       float64
t1_elders                       float64
t1_firstherald                  float64


In [12]:
cardinalidad = pd.concat([card_0, tipo_datos], axis=1)
cardinalidad.columns = ["uniques", "cardinalidad", "data_type"]

In [13]:
cardinalidad

uniques  cardinalidad data_type
gameId                           5148     99.980579    object
league                             23      0.446689    object
split                               7      0.135949    object
playoffs                            2      0.038842     int64
date                             5027     97.630608    object
game                                5      0.097106     int64
patch                              16      0.310740   float64
gamelength                       1403     27.248009     int64
t1_playerid                       244      4.738784    object
t1_ban1                           107      2.078073    object
t1_ban2                           109      2.116916    object
t1_ban3                           111      2.155758    object
t1_ban4                           128      2.485920    object
t1_ban5                           124      2.408235    object
t1_result                           2      0.038842     int64
t1_firstblood                       2      0.038842   float64
t1_firstdragon                      2      0.038842   float64
t1_dragons                          9      0.174791     int64
t1_elementaldrakes                  5      0.097106   float64
t1_infernals                        5      0.097106   float64
t1_mountains                        5      0.097106   float64
t1_clouds                           5      0.097106   float64
t1_oceans                           5      0.097106   float64
t1_elders                           3      0.058264   float64
t1_firstherald                      2      0.038842   float64
t1_heralds                          3      0.058264   float64
t1_firstbaron                       2      0.038842   float64
t1_barons                           5      0.097106     int64
t1_firsttower                       2      0.038842   float64
t1_towers                          12      0.233055     int64
t1_firstmidtower                    2      0.038842   float64
t1_firsttothreetowers               2      0.038842   float64
t1_inhibitors                       8      0.155370   float64
t2_playerid                       244      4.738784    object
t2_ban1                           101      1.961546    object
t2_ban2                           105      2.039231    object
t2_ban3                           106      2.058652    object
t2_ban4                           135      2.621868    object
t2_ban5                           125      2.427656    object
t2_result                           2      0.038842     int64
t2_firstblood                       2      0.038842   float64
t2_firstdragon                      2      0.038842   float64
t2_dragons                         10      0.194212     int64
t2_elementaldrakes                  5      0.097106   float64
t2_infernals                        5      0.097106   float64
t2_mountains                        5      0.097106   float64
t2_clouds                           5      0.097106   float64
t2_oceans                           5      0.097106   float64
t2_elders                           4      0.077685   float64
t2_firstherald                      2      0.038842   float64
t2_heralds                          3      0.058264   float64
t2_firstbaron                       2      0.038842   float64
t2_barons                           5      0.097106     int64
t2_firsttower                       2      0.038842   float64
t2_towers                          12      0.233055     int64
t2_firstmidtower                    2      0.038842   float64
t2_firsttothreetowers               2      0.038842   float64
t2_inhibitors                       8      0.155370   float64
t1p1_position                       1      0.019421    object
t1p1_player                       341      6.622645    object
t1p1_team                         244      4.738784    object
t1p1_champion                      51      0.990484    object
t1p1_kills                         10      0.194212     int64
t1p1_deaths                        12      0.233055     int64


In [14]:
cardinalidad.T

gameId    league     split  playoffs       date      game  \
uniques            5148        23         7         2       5027         5   
cardinalidad  99.980579  0.446689  0.135949  0.038842  97.630608  0.097106   
data_type        object    object    object     int64     object     int64   

                patch gamelength t1_playerid   t1_ban1   t1_ban2   t1_ban3  \
uniques            16       1403         244       107       109       111   
cardinalidad  0.31074  27.248009    4.738784  2.078073  2.116916  2.155758   
data_type     float64      int64      object    object    object    object   

              t1_ban4   t1_ban5 t1_result t1_firstblood t1_firstdragon  \
uniques           128       124         2             2              2   
cardinalidad  2.48592  2.408235  0.038842      0.038842       0.038842   
data_type      object    object     int64       float64        float64   

             t1_dragons t1_elementaldrakes t1_infernals t1_mountains  \
uniques               9                  5            5            5   
cardinalidad   0.174791           0.097106     0.097106     0.097106   
data_type         int64            float64      float64      float64   

             t1_clouds t1_oceans t1_elders t1_firstherald t1_heralds  \
uniques              5         5         3              2          3   
cardinalidad  0.097106  0.097106  0.058264       0.038842   0.058264   
data_type      float64   float64   float64        float64    float64   

             t1_firstbaron t1_barons t1_firsttower t1_towers t1_firstmidtower  \
uniques                  2         5             2        12                2   
cardinalidad      0.038842  0.097106      0.038842  0.233055         0.038842   
data_type          float64     int64       float64     int64          float64   

             t1_firsttothreetowers t1_inhibitors t2_playerid   t2_ban1  \
uniques                          2             8         244       101   
cardinalidad              0.038842       0.15537    4.738784  1.961546   
data_type                  float64       float64      object    object   

               t2_ban2   t2_ban3   t2_ban4   t2_ban5 t2_result t2_firstblood  \
uniques            105       106       135       125         2             2   
cardinalidad  2.039231  2.058652  2.621868  2.427656  0.038842      0.038842   
data_type       object    object    object    object     int64       float64   

             t2_firstdragon t2_dragons t2_elementaldrakes t2_infernals  \
uniques                   2         10                  5            5   
cardinalidad       0.038842   0.194212           0.097106     0.097106   
data_type           float64      int64            float64      float64   

             t2_mountains t2_clouds t2_oceans t2_elders t2_firstherald  \
uniques                 5         5         5         4              2   
cardinalidad     0.097106  0.097106  0.097106  0.077685       0.038842   
data_type         float64   float64   float64   float64        float64   

             t2_heralds t2_firstbaron t2_barons t2_firsttower t2_towers  \
uniques               3             2         5             2        12   
cardinalidad   0.058264      0.038842  0.097106      0.038842  0.233055   
data_type       float64       float64     int64       float64     int64   

             t2_firstmidtower t2_firsttothreetowers t2_inhibitors  \
uniques                     2                     2             8   
cardinalidad         0.038842              0.038842       0.15537   
data_type             float64               float64       float64   

             t1p1_position t1p1_player t1p1_team t1p1_champion t1p1_kills  \
uniques                  1         341       244            51         10   
cardinalidad      0.019421    6.622645  4.738784      0.990484   0.194212   
data_type           object      object    object        object      int64   

             t1p1_deaths t1p1_assists t1p1_firstblood t1p1_firstbloodkill  \
uniques        